### DISCUSSION TITLE
Add Array for loop firestore

### DISCUSSION THREAD

**Discussion by rzakfirst** [SchneiderInfosystems/FB4D#126] [Q&A] [answered]
Hello, Happy new year; I use TMemo to store all mutiliple files downlaod link for my uploaded images in firebase Stockage. Now I would like to tie his links to the post. To save the link I do it like this 

Memo4.Lines.Add(Obj.DownloadUrl);

Now to add them all in one post, I do it like this, but on firestore nothing.

jsonArray:= TJSONArray.Create;
 for I := 0 to Memo4.Lines.Count - 1 do
begin
   jsonArray.Add(Memo4.Lines.Strings[I]);
end;
Doc.AddOrUpdateField(TJSONPair.Create('Images', jsonArray));

Please help me!

**Comment by SchneiderInfosystems** [comment]
Do you want to store a document into the Firestore DB containing an array of links to the objects in the storage? 

Be aware that in the Firestore the JSON object must be formatted according to the Firestore data types.
If I do understand your task properly, the following change solves this formatting issue:

```
Doc.AddOrUpdateField(TJSONObject.SetArray('Images',[
   TJSONObject.SetStringValue(Memo4.Lines[0], 
   TJSONObject.SetStringValue(Memo4.Lines[1], .. 
   TJSONObject.SetStringValue(Memo4.Lines[Memo4.Lines.Count - 1]])));
```

For more information please check the [wiki about writing documents into the Firestore DB](https://github.com/SchneiderInfosystems/FB4D/wiki/FB4D-Reference-IFirestoreDocument#write-access).

**Reply by rzakfirst** [reply]
And if the user add more or less images than the defined lines. Because if I do jDoc.AddOrUpdateField (TJSONObject.SetArray ('Images', [
    TJSONObject.SetStringValue (Memo4.Lines [0],
    TJSONObject.SetStringValue (Memo4.Lines [1], ..
    TJSONObject.SetStringValue (Memo4.Lines [Memo4.Lines.Count - 1]])));

It is as if the number of image links is predefined. One can add one or two, the other three, and so on.

**Comment by rzakfirst** [comment, marked as answer]
I am in the process of designing a mobile application that allows users to post announcements, in which they can add one or more images. And I would like all the links of the added images to be attached with the post. So during upload, each image uploaded successfully, its link is added on a line in TMemo, and once all the images are uploaded, I would like all the links (which are on the TMemo lines) to be added as Array. The total of Tmemo lines is not constant. Users can add one or more images.
![Screenshot_20220105-235555_Chrome](https://user-images.githubusercontent.com/57506168/148301500-136796f7-6afc-4293-8d0c-0b72588211f9.jpg)
I would like links that are on the TMemo Lines to be added like this
![Screenshot_20220105-235835_Chrome](https://user-images.githubusercontent.com/57506168/148301772-3487ac76-cc22-4fca-adaa-84e804004c22.jpg)

**Reply by SchneiderInfosystems** [reply]
Unfortunately, I see the last post only today. But this is simple. A loop over the lines of the memo field and append to the array, which is passed to the `SetArray`.

**Reply by rzakfirst** [reply]
How to do it, please?

**Reply by SchneiderInfosystems** [reply]
Please consider my policy for free support: https://github.com/SchneiderInfosystems/FB4D/discussions/53#discussioncomment-265330

**Reply by SchneiderInfosystems** [reply]
Today implemented for my project:
```
var
  Arr: array of TJSONValue;
  s: string;
  sl: TStringList;
  c: integer;
begin
  SetLength(Arr, sl.Count); c:= 0;
  for s in sl do
  begin
      Arr[c] := TJSONObject.SetStringValue(s);
      inc(c);
  end;
  fDocument.AddOrUpdateField(TJSONObject.SetArray('MyArry', Arr));
end;
```
